### PR TITLE
Ultralytics direct integration documentation

### PR DESCRIPTION
**Pull request by bfineran**

### CODE REVIEW

**Review by markurtz — CHANGES_REQUESTED** [review]
I haven't looked through the docs yet, will do a bit later today. But, we do need to put a new section in the index.rst for Integrations and have this as one of the links under it

File: docs/integrations/ultralytics.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+| ----------- | ----------- | ----------- |
+| YOLOv3-pruned | 88% sparse YOLOv3-SPP model | "zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned-aggressive_97" |
+| YOLOv3-pruned_quant | 83% sparse YOLOv3-SPP model with INT8 quantization | "zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned_quant-aggressive_94" |
+
+
+### Other arguments
+When a recipe is passed in to `--sparseml-recipe`, a `ScheduledModifierManager`
+and `ScheduledOptimizer` will be created to modify the normal training flow for
+sparsification. For best results with SparseML the following arguments may also
+have to be set.  In generall, all should be set when running with SparseML
```
Comment (markurtz) [inline docs/integrations/ultralytics.md:42]: typo: generall

**Comment by bfineran** [comment]
@markurtz added a section in `index.rst` for integrations. let me know if this works/is what you had in mind

**Review by markurtz — DISMISSED** [review]

File: docs/integrations/ultralytics.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+The following SparseZoo stubs can be used directly with `--sparseml-recipes` to load
+their associated recipes
+
+| Model | Description | SparseZoo Stub |
+| ----------- | ----------- | ----------- |
+| YOLOv3-pruned | 88% sparse YOLOv3-SPP model | "zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned-aggressive_97" |
+| YOLOv3-pruned_quant | 83% sparse YOLOv3-SPP model with INT8 quantization | "zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned_quant-aggressive_94" |
+
+
+### Other arguments
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:38]:
```suggestion
### Other Arguments
```

File: docs/integrations/ultralytics.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+When a recipe is passed in to `--sparseml-recipe`, a `ScheduledModifierManager`
+and `ScheduledOptimizer` will be created to modify the normal training flow for
+sparsification. For best results with SparseML the following arguments may also
+have to be set.  In general, all should be set when running with SparseML
+(with the exception of `--disable-amp` if not using quantization).
+
+* `--export-onnx` - Exports the final model to ONNX after training completes. Required
+    for inference with the [DeepSparse](https://github.com/neuralmagic/deepsparse)
+    engine
+* `--use-leaky-relu` - LeakyReLU runs best on the DeepSparse engine. Setting this flag
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:48]:
```suggestion
* `--use-leaky-relu` - LeakyReLU runs best on the DeepSparse Engine. Setting this flag
```

File: docs/integrations/ultralytics.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+SparseZoo as follows.  The model achieves 64.2 mAP@0.5.
+
+```python
+from sparsezoo.models.detection import yolo_v3
+
+base_model = yolo_v3()
+print(f"downloaded path: {base_model.framework_files[0].downloaded_path()}")
+```
+
+After downloading, the weights can be moved to a file locatin of your choice:
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:68]:
```suggestion
After downloading, the weights can be moved to a file location of your choice:
```

File: docs/integrations/ultralytics.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+```bash
+cp <DOWNLOADED-PATH> <YOUR-LOCAL-PATH>
+```
+
+### Example command
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:73]:
```suggestion
### Example Command
```

File: docs/integrations/ultralytics.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  --data coco.yaml \
+  --hyp data/hyp.finetune.yaml \
+  --epochs 242 \
+  --batch-size 48 \
+  --name yolov3-spp-leaky_relu-pruned_quant
+```
+
+### Benchmarking and Examples
+For examples of benchmarking and deploying sparsified YOLO models with the DeepSparse
+engine, take a look at the `deepsparse/` examples
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:92]:
```suggestion
Engine, take a look at the `deepsparse/` examples
```

File: docs/integrations/ultralytics.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+### Other arguments
+When a recipe is passed in to `--sparseml-recipe`, a `ScheduledModifierManager`
+and `ScheduledOptimizer` will be created to modify the normal training flow for
+sparsification. For best results with SparseML the following arguments may also
+have to be set.  In general, all should be set when running with SparseML
+(with the exception of `--disable-amp` if not using quantization).
+
+* `--export-onnx` - Exports the final model to ONNX after training completes. Required
+    for inference with the [DeepSparse](https://github.com/neuralmagic/deepsparse)
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:46]:
```suggestion
    for inference with the [DeepSparse Engine](https://github.com/neuralmagic/deepsparse)
```

File: docs/integrations/ultralytics.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    engine
```
Comment (jeanniefinks) [inline docs/integrations/ultralytics.md:47]:
```suggestion

```

**Review by markurtz — APPROVED** [review]

**Comment by bfineran** [comment]
closing since the integration has now moved to a public fork of the `yolov3` repo and our integrations structure has changed.

**Pull request closed without merging**